near-sdk/src/collections/tree_map.rs

```diff
@@ -0,0 +1,1637 @@
+        assert_eq!(map.height(), 0);
+        assert_eq!(map.get(&42), None);
+        assert!(!map.contains_key(&42));
+        assert_eq!(map.min(), None);
+        assert_eq!(map.max(), None);
+        assert_eq!(map.ceil(&42), None);
+        assert_eq!(map.floor(&42), None);
+    }
+
+    #[test]
+    fn test_insert_3_rotate_l_l() {
+        test_env::setup();
+
+        let mut map: TreeMap<i32, i32> = TreeMap::default();
+        assert_eq!(map.height(), 0);
+
+        map.insert(&3, &3);
+        assert_eq!(map.height(), 1);
+
+        map.insert(&2, &2);
+        assert_eq!(map.height(), 2);
+
+        map.insert(&1, &1);
+
+        let root = map.root;
+        assert_eq!(root, 1);
+        assert_eq!(map.key.get(&root), Some(2));
+        assert_eq!(map.height(), 2);
+
+        map.clear();
+    }
+
+    #[test]
+    fn test_insert_3_rotate_r_r() {
+        test_env::setup();
+
+        let mut map: TreeMap<i32, i32> = TreeMap::default();
+        assert_eq!(map.height(), 0);
+
+        map.insert(&1, &1);
+        assert_eq!(map.height(), 1);
+
+        map.insert(&2, &2);
+        assert_eq!(map.height(), 2);
+
+        map.insert(&3, &3);
+
+        let root = map.root;
+        assert_eq!(root, 1);
+        assert_eq!(map.key.get(&root), Some(2));
+        assert_eq!(map.height(), 2);
+
+        map.clear();
+    }
+
+    #[test]
+    fn test_insert_lookup_n_asc() {
+        test_env::setup();
+
+        let mut map: TreeMap<i32, i32> = TreeMap::default();
+
+        let n: u64 = 30;
+        let cases = (0..2*(n as i32)).collect::<Vec<i32>>();
+
+        let mut counter  = 0;
+        for k in &cases {
+            if *k % 2 == 0 {
+                counter += 1;
+                map.insert(k, &counter);
+            }
+        }
+
+        counter = 0;
+        for k in &cases {
+            if *k % 2 == 0 {
+                counter += 1;
+                assert_eq!(map.get(k), Some(counter));
+            } else {
+                assert_eq!(map.get(k), None);
+            }
+        }
+
+        assert!(map.height() <= max_tree_height(n));
+        map.clear();
+    }
+
+    #[test]
+    fn test_insert_lookup_n_desc() {
+        test_env::setup();
+
+        let mut map: TreeMap<i32, i32> = TreeMap::default();
+
+        let n: u64 = 30;
+        let cases = (0..2*(n as i32)).rev().collect::<Vec<i32>>();
+
+        let mut counter  = 0;
+        for k in &cases {
+            if *k % 2 == 0 {
+                counter += 1;
+                map.insert(k, &counter);
+            }
+        }
+
+        counter = 0;
+        for k in &cases {
+            if *k % 2 == 0 {
+                counter += 1;
+                assert_eq!(map.get(k), Some(counter));
+            } else {
+                assert_eq!(map.get(k), None);
+            }
+        }
+
+        assert!(map.height() <= max_tree_height(n));
+        map.clear();
+    }
+
+    #[test]
+    fn insert_n_random() {
+        test_env::setup();
+
+        for k in 1..5 {
```

Comment:
5 => 500? 5 is just not serious :)

Reply:
I can't agree more! Unfortunately, even with `1..6` results in `HostError(GasLimitExceeded)`. Any suggestions how to address it?

Reply:
When we create `MockedBlockchain` in `test_env::setup()` we pass `Default::default()` as a `VMConfig`. In this specific test instead of calling `test_env::setup();` we can unwrap that code and initialize `VMConfig` with `limit_config::max_gas_burnt` equal to `u64::MAX`.

Also, @SkidanovAlex `5` is not the number of items we insert in the tree here, `let n = 1 << k;` is.